components/tidb_query_datatype/src/codec/batch/lazy_column.rs

```diff
@@ -224,7 +224,7 @@ impl LazyBatchColumn {
     pub fn encode(
         &self,
         row_index: usize,
-        field_type: &FieldType,
+        field_type: &dyn FieldTypeAccessor,
```

Comment:
Prefer `impl FieldTypeAccessor` over `dyn` for performance considerations.

Reply:
If we use `impl FieldTypeAccessor` instead, the `ft` should be move from the context which will cause some extra cloning of `ColumnInfo` or `FieldType`

Reply:
Done.